Unstable test TestPoolTuneScaleUpAndDown

## Bug Report

```
[8,378 / 8,631] 229 / 266 tests; Testing //resourcemanager/pool/spool:spool_test (shard 2 of 2); 299s remote-cache, linux-sandbox ... (140 actions running)



TIMEOUT: //resourcemanager/pool/spool:spool_test (Summary)

      /home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/execroot/__main__/bazel-out/k8-fastbuild/testlogs/resourcemanager/pool/spool/spool_test/shard_2_of_2/test.log

      /home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/execroot/__main__/bazel-out/k8-fastbuild/testlogs/resourcemanager/pool/spool/spool_test/shard_2_of_2/test_attempts/attempt_1.log

      /home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/execroot/__main__/bazel-out/k8-fastbuild/testlogs/resourcemanager/pool/spool/spool_test/shard_2_of_2/test_attempts/attempt_2.log

ERROR: /home/jenkins/agent/workspace/pingcap/tidb/ghpr_unit_test/tidb/resourcemanager/pool/spool/BUILD.bazel:25:8: Testing //resourcemanager/pool/spool:spool_test (shard 2 of 2) failed: Test failed, aborting

==================== Test output for //resourcemanager/pool/spool:spool_test (shard 2 of 2):

=== RUN   TestPoolTuneScaleUpAndDown

    spool_test.go:101: expect 8 workers running, but got 6

    spool_test.go:109: 

        	Error Trace:	resourcemanager/pool/spool/spool_test.go:109

        	Error:      	Not equal: 

        	            	expected: 2

        	            	actual  : 0

        	Test:       	TestPoolTuneScaleUpAndDown

--- FAIL: TestPoolTuneScaleUpAndDown (0.70s)

=== RUN   TestRunWithNotEnough

--- PASS: TestRunWithNotEnough (1.00s)

FAIL

coverage: 2.3% of statements

--

Coverage runner: Not collecting coverage for failed test.

The following commands failed with status 1

/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/linux-sandbox/5259/execroot/__main__/bazel-out/k8-fastbuild/bin/resourcemanager/pool/spool/spool_test_/spool_test.runfiles/__main__/resourcemanager/pool/spool/spool_test_/spool_test

================================================================================

==================== Test output for //resourcemanager/pool/spool:spool_test (shard 2 of 2):

=== RUN   TestPoolTuneScaleUpAndDown

    spool_test.go:101: expect 8 workers running, but got 7

    spool_test.go:109: 

        	Error Trace:	resourcemanager/pool/spool/spool_test.go:109

        	Error:      	Not equal: 

        	            	expected: 2

        	            	actual  : 1

        	Test:       	TestPoolTuneScaleUpAndDown

--- FAIL: TestPoolTuneScaleUpAndDown (0.70s)

=== RUN   TestRunWithNotEnough

--- PASS: TestRunWithNotEnough (1.00s)

FAIL

coverage: 2.3% of statements

--

Coverage runner: Not collecting coverage for failed test.

The following commands failed with status 1

/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/linux-sandbox/5317/execroot/__main__/bazel-out/k8-fastbuild/bin/resourcemanager/pool/spool/spool_test_/spool_test.runfiles/__main__/resourcemanager/pool/spool/spool_test_/spool_test

================================================================================

==================== Test output for //resourcemanager/pool/spool:spool_test (shard 2 of 2):

=== RUN   TestPoolTuneScaleUpAndDown

    spool_test.go:101: expect 8 workers running, but got 5

-- Test timed out at 2023-03-13 18:20:03 CST --

================================================================================

ERROR: Test failed, aborting

INFO: Elapsed time: 364.274s, Critical Path: 340.02s

INFO: 8436 processes: 5192 remote cache hit, 1121 internal, 2123 linux-sandbox.

INFO: Build completed, 1 test FAILED, 8436 total actions

Test cases: finished with 5480 passing and 1 failing out of 5481 test cases



Executed 80 out of 266 tests: 229 tests pass, 1 fails locally and 36 were skipped.
```

--- a/resourcemanager/pool/spool/spool_test.go
+++ b/resourcemanager/pool/spool/spool_test.go
@@ -96,7 +96,7 @@ func TestPoolTuneScaleUpAndDown(t *testing.T) {
 		}()
 	}
 	p.Tune(8)
-	time.Sleep(500 * time.Millisecond)
+	wg.Wait()
 	if n := p.Running(); n != 8 {
 		t.Errorf("expect 8 workers running, but got %d", n)
 	}
@@ -119,7 +119,6 @@ func TestPoolTuneScaleUpAndDown(t *testing.T) {
 		cnt.Add(1)
 	}
 	fnChan := make(chan func(), 10)
-	wg.Wait()
 	err := p.RunWithConcurrency(fnChan, 2)
 	require.NoError(t, err)
 	require.Equal(t, int32(2), p.Running())

